test/blackbox/run_blackbox_tests/test_cases_encrypted_storage.c

```diff
@@ -738,6 +738,575 @@ static bool test_steps_encrypted_storage_08(void) {
+	assert(read(pipefd[0], invitation_buff, 200) != -1);
+
+	// Open peer and NUT node instances and join peer with the obtained invitation.
+
+	mesh = meshlink_open_encrypted("encrypted_conf.8" , "nut", "test", DEV_CLASS_BACKBONE, key, keysize);
+	assert(mesh);
+	mesh1 = meshlink_open_encrypted("encrypted_conf.9" , "peer", "test", DEV_CLASS_BACKBONE, key, keysize);
+	assert(mesh1);
+	assert(meshlink_start(mesh));
+	bool join_status = meshlink_join(mesh1, invitation_buff);
+	assert_int_equal(join_status, true);
+	node = meshlink_get_node(mesh, "peer");
+	assert_int_not_equal(node, NULL);
+
+	// Cleanup
+
+	meshlink_close(mesh);
+	meshlink_close(mesh1);
+	meshlink_destroy("encrypted_conf.8");
+	meshlink_destroy("encrypted_conf.9");
+
+	return true;
+}
+
+/* Execute encrypted storage Test Case # 15 - API synchronization tests */
+static void test_case_encrypted_storage_15(void **state) {
+	execute_test(test_steps_encrypted_storage_15, state);
+}
+
+/* Test Steps for encrypted storage Test Case # 15
+    @ meshlink_set_port @ sync test
+
+    Test Steps:
+    1. In the child process open NUT node, set new meshlink listening port and terminate
+        immediately on API returning by sending SIGINT.
+    2. Test driver reopens node and get meshlink listening port
+
+    Expected Result:
+    Meshlink should see the new port that's being set as node's listening.
+*/
+static bool test_steps_encrypted_storage_15(void) {
+	char key[] = "encryptionkey";
+	int keysize = sizeof(key);
+	int child_ret;
+	pid_t pid;
+	meshlink_handle_t *mesh;
+
+	meshlink_destroy("encrypted_conf.8");
+
+	// Create node instance in a separate child process
+
+	pid = fork();
+	assert(pid != -1);
+
+	if(pid == 0) {
+
+		// Open NUT node instance and set a new listening port number
+		mesh = meshlink_open_encrypted("encrypted_conf.8" , "nut", "test", DEV_CLASS_BACKBONE, key, keysize);
+		assert(mesh);
+		bool set_port_status = meshlink_set_port(mesh, 1111);
+		assert(set_port_status);
+		assert(!raise(SIGINT));
+		sleep(1);
+		abort();
+	}
+
+	assert(waitpid(pid, &child_ret, 0) != -1);
+	assert(WIFSIGNALED(child_ret) && (WTERMSIG(child_ret) == SIGINT));
+
+	// Reopen NUT node
+
+	mesh = meshlink_open_encrypted("encrypted_conf.8" , "nut", "test", DEV_CLASS_BACKBONE, key, keysize);
+	assert(mesh);
+
+	// Obtain meshlink listening port and verify the port being set using @ meshlink_set_port @
+	// succeeded or not
+
+	int port = meshlink_get_port(mesh);
+	assert_int_equal(port, 1111);
+
+	// Cleanup
+
+	meshlink_close(mesh);
+	meshlink_destroy("encrypted_conf.8");
+	return true;
+}
+
+/* Execute encrypted storage Test Case # 16 - API synchronization tests */
+static void test_case_encrypted_storage_16(void **state) {
+	execute_test(test_steps_encrypted_storage_16, state);
+}
+
+/* Test Steps for encrypted storage Test Case # 16
+    @ meshlink_set_canonical_address @ sync test
+
+    Test Steps:
+    1. In the child process open NUT node, set new canonical address and terminate
```

Comment:
That's very similar to what test case #15 does. Maybe these two can be combined, and in step 1 set both a canonical address and call `meshlink_set_port()`, and in step 2 check the result of `meshlink_get_port()` and that an invitation contains the canonical address?

Reply:
Correct but I thought the goal is to test per API. Actually `meshlink_set_port( )` persists the port number but `meshlink_set_canonical_address( )` actually don't write the address into the secondary storage.

Reply:
Ah, I was thinking it would be more efficient to combine them, but since it's a test suite it's not that important. If you think it's cleaner to keep the test cases separate, then that's fine.